### PR TITLE
adds InteractionOperator utilities

### PR DESCRIPTION
**Pull request by bryano**
handled by normal_ordered, is_hermitian, and hermitian_conjugate

new zero construction method

new projection method

### CODE REVIEW

File: src/openfermion/utils/_operator_utils.py [hunk 1]
```diff
@@ -839,9 +849,36 @@ def normal_ordered(operator, hbar=1.):
         order_fn = normal_ordered_quad_term
         kwargs['hbar'] = hbar
 
+    elif isinstance(operator, InteractionOperator):
```
Comment (kevinsung) [inline src/openfermion/utils/_operator_utils.py]: What does it mean to normal-order an InteractionOperator?

Comment (bryano) [reply src/openfermion/utils/_operator_utils.py]: The docstring for `normal_ordered` says that, in addition to putting creation operators before annihilation operators, it orders the operators in each set by index. `InteractionOperator` takes care of the former by construction; this does the latter.

Comment (bryano) [reply src/openfermion/utils/_operator_utils.py]: This actually orders from lowest to highest, so I'll need to either update the docstring or change the code. Why the convention that the modes are ordered from highest to lowest?

Comment (kevinsung) [reply src/openfermion/utils/_operator_utils.py]: I don't know why that convention was picked. In any case, let's stick with the old convention.

Comment (bryano) [reply src/openfermion/utils/_operator_utils.py]: Done

Comment (bryano) [reply src/openfermion/utils/_operator_utils.py]: @kevinsung ready for re-review

File: src/openfermion/utils/_operator_utils.py [hunk 2]
```diff
@@ -238,6 +237,26 @@ def hermitian_conjugated(operator):
...
+            operator.one_body_tensor)
+        conjugate_two_body_tensor = hermitian_conjugated(
+            operator.two_body_tensor)
+        conjugate_operator = type(operator)(conjugate_constant,
+            conjugate_one_body_tensor, conjugate_two_body_tensor)
+
+    # Handle dicts
+    # Assumes key corresponds to indices of some number of creation operators
+    # followed by the same number of annihilation operators.
+    elif isinstance(operator, dict):
```
Comment (kevinsung) [inline src/openfermion/utils/_operator_utils.py]: It's not our place to define the hermitian conjugate of a dictionary; delete this.

File: src/openfermion/ops/_polynomial_tensor.py [hunk 3]
```diff
@@ -221,11 +223,17 @@ def __add__(self, addend):
...
-            neg_n_body_tensors[key] = numpy.negative(self.n_body_tensors[key])
-        return PolynomialTensor(neg_n_body_tensors)
+            new_n_body_tensors[key] = func(self.n_body_tensors[key])
+        return PolynomialTensor(new_n_body_tensors)
+
+    def __neg__(self):
+        return self.with_function_applied_elementwise(operator.neg)
+
+    def __mod__(self, other):
+        return self.with_function_applied_elementwise(lambda x: x % other)
```
Comment (kevinsung) [inline src/openfermion/ops/_polynomial_tensor.py]: Do you have an actual use case for modding a PolynomialTensor? Seems quite unnatural to me ...

File: src/openfermion/utils/_operator_utils.py (hunk 2 above)
Comment (bryano) [reply src/openfermion/utils/_operator_utils.py]: Done

File: src/openfermion/ops/_polynomial_tensor.py (hunk 3 above)
Comment (bryano) [reply src/openfermion/ops/_polynomial_tensor.py]: Yes, I have some code that exponentiates an InteractionOperator that I test by comparing it the the log of the exponentiated unitary.

File: src/openfermion/utils/_operator_utils_test.py [hunk 4]
```diff
@@ -403,6 +405,14 @@ def test_hermitian_conjugated_semihermitian(self):
                  BosonOperator('2^ 2', -0.1j))
         self.assertEqual(op_hc, hermitian_conjugated(op))
 
+    def test_hermitian_conjugated_interaction_operator_and_dict(self):
```
Comment (kevinsung) [inline src/openfermion/utils/_operator_utils_test.py]: Remove `_and_dict`

Comment (bryano) [reply src/openfermion/utils/_operator_utils_test.py]: Done

**Comment by bryano** [comment]
@kevinsung I think this is good to go

**Comment by bryano** [comment]
@kevinsung Could you merge this?

**Pull request merged**